buildbot/master/files/config/factories.py

```diff
@@ -103,6 +104,95 @@ def make_step(self, command):
         return step_class(**step_kwargs)
 
 
+class StepsYAMLParsingStep(buildstep.ShellMixin, buildstep.BuildStep):
+    """
+    Step which resolves the in-tree YAML and dynamically add test steps.
+    """
+
+    haltOnFailure = True
+    flunkOnFailure = True
+
+    def __init__(self, builder_name, environment, yaml_path):
+        self.builder_name = builder_name
```

Comment:
I believe you also need to setup the shell mixin and call `BuildStep.__init__` as done in the example in the Buildbot Nine docs.